decidim/decidim
# Accessibility tool for development environments

#### :tophat: What? Why?
This implements a visual accessibility evaluation tool for the `decidim-dev` module that can be used to decrease the amount of accessibility issues in the views.

The tool becomes available when the following is true:
1. The `decidim-dev` has been defined as a dependency in the Gemfile (generally in the `:development` group)
2. The app is running in the `development` environment (i.e. `RAILS_ENV=development` which is the default)

This also adds the tool to the design app.

#### :pushpin: Related Issues
- Related to #7764, #6207

#### Testing
1. Start the application in the development mode
2. See the accessibility tool badge appear at the top left corner of the page (see screenshots)

#### :clipboard: Checklist


### :camera: Screenshots

The tool in action:

https://user-images.githubusercontent.com/864340/112805435-40a89080-907e-11eb-972c-cfa26dc7fea3.mp4

It displays a `WAI WCAG` badge at the left side of the screen and clicking that opens a list of accessibility violations on that page.

When everything is OK it looks like this:
![WAI WCAG badge when everything is OK](https://user-images.githubusercontent.com/864340/112807519-adbd2580-9080-11eb-9f64-9d0795c8e883.png)

When there are errors on the page, it looks like this (the number indicates the different violation categories on the page):
![WAI WCAG badge when there are violations on the page](https://user-images.githubusercontent.com/864340/112807412-8b2b0c80-9080-11eb-9eed-e4a2e4b60d96.png)

When the sidebar is opened, the sections of the page that have violations are highlighted on the page like this:
![Accessibility violation indicators on the page](https://user-images.githubusercontent.com/864340/112807716-e230e180-9080-11eb-84da-2c478e351094.png)

When you click the violation indicator (:warning:), it will highlight which issue this is as listed in the sidebar (shown in the video).
